### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by Danathar** [osxfuse/sshfs#82]
Given the last commit was in 2017, I'm wondering if there is any updates? I was going to load the software but I'm a but hesitant to load sshfs if there is no maintainer.

**Comment by SevDaisy** [comment]
1. Is this project dead? I can't agree with that. 
2. Look at other issue such as #81 ，it can be responded at a short time.
3. poor english of mine, sorry

**Comment by huuuuuuuuuuu** [comment]
Can anyone suggest an alternative?

**Comment by n-rodriguez** [comment]
> Can anyone suggest an alternative?

I don't think there is one. Actually the most important part of this project is maintained https://github.com/osxfuse/osxfuse. You can still compile `sshfs` on your machine (with a small change).

```diff
diff --git a/sshfs.c b/sshfs.c
index f116633..714317b 100644
--- a/sshfs.c
+++ b/sshfs.c
@@ -14,9 +14,6 @@
 #if !defined(__CYGWIN__)
 #include <fuse_lowlevel.h>
 #endif
-#ifdef __APPLE__
-#  include <fuse_darwin.h>
-#endif
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
```

If you speak japanese you can use this custom homebrew formula : https://hkob.hatenablog.com/entry/2020/12/03/110000

**Comment by sbthandras** [comment]
> > Can anyone suggest an alternative?
> 
> I don't think there is one. Actually the most important part of this project is maintained https://github.com/osxfuse/osxfuse. You can still compile `sshfs` on your machine (with a small change).
> 
> ```diff
> diff --git a/sshfs.c b/sshfs.c
> index f116633..714317b 100644
> --- a/sshfs.c
> +++ b/sshfs.c
> @@ -14,9 +14,6 @@
>  #if !defined(__CYGWIN__)
>  #include <fuse_lowlevel.h>
>  #endif
> -#ifdef __APPLE__
> -#  include <fuse_darwin.h>
> -#endif
>  #include <assert.h>
>  #include <stdio.h>
>  #include <stdlib.h>
> ```
> 
> If you speak japanese you can use this custom homebrew formula : https://hkob.hatenablog.com/entry/2020/12/03/110000

Well the thing is it does "work" for me, but it is for some reason way slower than VScode's Remote SSH editing when viewing files, and if I ever lose connection if drags the whole Finder with it, so it's very painful for me to work with

**Comment by mrdc** [comment]
> Given the last commit was in 2017, I'm wondering if there is any updates? I was going to load the software but I'm a but hesitant to load sshfs if there is no maintainer.

There is a well maintained alternative and it's in README of this project: https://github.com/libfuse/sshfs

**Comment by easp** [comment]
@mrdc From the README of that project:

> However, at present SSHFS does not have any active, regular contributors, and there are a number of known issues

If you are referring to the [corresponding homebrew package](https://formulae.brew.sh/formula/sshfs) it is marked as depreciated. There is only an x86_64 build.

Attempting to `brew install sshfs` results in this message:
```
Warning: sshfs has been deprecated because it has an archived upstream repository!
sshfs: Linux is required for this software.
libfuse: Linux is required for this software.
Error: sshfs: Unsatisfied requirements failed this build.
```
So, what, specifically are you referring to?

**Comment by mrdc** [comment]
> From the README of that project:

In README of osxfuse/sshfs:
>The latest version and more information can be found on http://github.com/libfuse/sshfs

This version is actively developed and has latest commits one month ago.

**Comment by verdverm** [comment]
@mrdc, from the latest release notes on that project (https://github.com/libfuse/sshfs/releases)

> SSHFS is now no longer maintained or developed

**Comment by mrdc** [comment]
> @mrdc, from the latest release notes on that project (https://github.com/libfuse/sshfs/releases)
> 
> > SSHFS is now no longer maintained or 
developed


The latest merge was made last week.
For me it's stable, so no freq changes are necessary.

No longer maintained by the original maintainer.
There are other people to do it - it's open source and that's how it works.

**Comment by easp** [comment]
@mrdc. libfuse/sshfs requires libfuse 3.1.0 or newer or another implementation of the libfuse3 interface. How did you get it to build and run on MacOS?

**Comment by marco-brandizi** [comment]
Hi all,
my experience with this:

* FUSE can be installed on mac via: 
  `brew install homebrew/cask-versions/macfuse-dev`
* If you install `brew install sshfs@2.10_2`, it should work with macfuse-dev (didn't try, since I had it already installed). Maybe higher versions of sshfs works too, but `brew install sshfs` (which picks the last version, 3.7.3 at the time of writing) doesn't work, with the messages:
  ```
  Warning: sshfs has been deprecated because it has an archived upstream repository!
  sshfs: Linux is required for this software.
  libfuse: Linux is required for this software.
  Error: sshfs: Unsatisfied requirements failed this build.
  ```
  **I don't understand the warning**, since the sshfs repo isn't archived.

I understand that actually I should install https://github.com/osxfuse/sshfs, but this isn't available in Homebrew.

Could someone publish it there? Also, this is a sub-project that looks dead, since it hasn't released anything since 2.5.0 in 2014, while it's Linux counterpart is at 3.7.3. If it's a fork, but I'd like to know if going backwards is worthwhile (and is the fork needed?).

Thanks for helping.

**Comment by bfleischer** [comment]
sshfs 3.x requires libfuse 3, but macFUSE ships with libfuse 2. Until libfuse 3 has been ported to macOS we are stuck with sshfs 2.x on macOS. As of writing this the latest version of sshfs that will run on macOS is version 2.10 which is serveral years old but works just fine as far as I can tell. A universal sshfs 2.10 build can be downloaded from https://macfuse.io.

macOS specific changes from this repository have been merged into the upstream repository https://github.com/libfuse/sshfs years ago. Future development will continue there.
